### PR TITLE
appsec: allow enabling sca from configuration

### PR DESCRIPTION
**Pull request by Julio-Guerra**
### What does this PR do?

Allow enabling ASM's Software Composition Analysis (ASM SCA) with the service configuration env var `DD_APPSEC_SCA_ENABLED`.

This is achieved from a new `init()` function in `internal/appsec/config` to set it at init-time of the app, and thanks to a new public API of the client to register app-wide configuration values like this one.

The rest of the code is mainly tied to the testability of the feature.

### Motivation

1. Deployment-time enablement, as opposed to the UI-based service per service one.
2. Bulk [deploy-time] enablement.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/appsec/config/config.go [hunk 1]
```diff
@@ -12,11 +12,40 @@ import (
...
+	if err != nil {
+		log.Error("appsec: %v", err)
+		return
+	}
+	if defined {
+		client.RegisterAppConfig(SCAEnabledEnv, val, "env_var")
+	}
+}
+
 // EnvEnabled is the env var used to enable/disable appsec
```
Comment (Hellzy) [inline internal/appsec/config/config.go:44]:
```suggestion
// The following environment variables dictate the enablement of ASM and some of its features
```

Comment (Julio-Guerra) [reply internal/appsec/config/config.go:44]: done in caf8f554

File: internal/appsec/config/config.go [hunk 2]
```diff
@@ -12,11 +12,40 @@ import (
...
-const EnvEnabled = "DD_APPSEC_ENABLED"
+const (
+	EnvEnabled    = "DD_APPSEC_ENABLED"
+	SCAEnabledEnv = "DD_APPSEC_SCA_ENABLED"
```
[the remark this answers is not in the document]
Comment (Julio-Guerra) [reply internal/appsec/config/config.go:47]: done in caf8f554

File: internal/appsec/config/config_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+package config
+
+import (
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry/telemetrytest"
+)
+
+func TestSCAEnable(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Julio-Guerra) [reply internal/appsec/config/config_test.go]: Fixed in c4442e06

**Review by Hellzy — APPROVED** [review]

**Pull request merged**